hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/wal/WALCellCodec.java

```diff
@@ -241,10 +246,27 @@ public void write(Cell cell) throws IOException {
         compression.getDictionary(CompressionContext.DictionaryIndex.FAMILY));
       PrivateCellUtil.compressQualifier(out, cell,
         compression.getDictionary(CompressionContext.DictionaryIndex.QUALIFIER));
-      // Write timestamp, type and value as uncompressed.
+      // Write timestamp, type and value.
       StreamUtils.writeLong(out, cell.getTimestamp());
-      out.write(cell.getTypeByte());
-      PrivateCellUtil.writeValue(out, cell, cell.getValueLength());
+      byte type = cell.getTypeByte();
+      if (compression.getValueCompressor() != null &&
+          cell.getValueLength() > VALUE_COMPRESS_THRESHOLD) {
+        // Try compressing the cell's value
+        byte[] compressedBytes = compressValue(cell);
+        // Only write the compressed value if we have achieved some space savings.
+        if (compressedBytes.length < cell.getValueLength()) {
+          // Set the high bit of type to indicate the value is compressed
+          out.write((byte)(type|0x80));
```

Comment:
> Jetty settled on a size threshold of 23 bytes.

Thank you @ndimiduk . gzip and deflate are the same thing, essentially. Let's opt for the smaller threshold and see how it goes. Worst case if the compressor produces output that is larger than the original, we just discard it and use the original, so that's not a problem. With a smaller threshold more values are eligible for compression so there will be more time spent in compression, but presumably with a pay off in space savings, so that seems fine.

Reply:
It is simpler if we don't try to conditionally compress values. No high bit twiddling. So there is no leakage into KeyValue in this case and no magic values. 

There also doesn't need to be a size threshold. Since we are using a single deflator instance over all values in the WAL file, overheads of too small values should be amortized.

Reply:
This was also a hold over from the initial version that had a Inflater/Deflater instance per value, so it would have made a difference then.